File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-cd-5-4/SingleValuationDate.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class SingleValuationDate : ISerialized
    {
        public SingleValuationDate(XmlNode xmlNode)
        {
            XmlNode businessDaysNode = xmlNode.SelectSingleNode("businessDays");
            
            if (businessDaysNode != null)
            {
                if (businessDaysNode.Attributes["href"] != null || businessDaysNode.Attributes["id"] != null) 
                {
                    if (businessDaysNode.Attributes["id"] != null) 
                    {
                        businessDaysIDRef_ = businessDaysNode.Attributes["id"].Value;
                        XsdTypeNonNegativeInteger ob = new XsdTypeNonNegativeInteger(businessDaysNode);
                        IDManager.SetID(businessDaysIDRef_, ob);
                    }
                    else if (businessDaysNode.Attributes["href"] != null)
                    {
                        businessDaysIDRef_ = businessDaysNode.Attributes["href"].Value;
                    }
                    else
                    {
                        businessDays_ = new XsdTypeNonNegativeInteger(businessDaysNode);
                    }
                }
                else
                {
                    businessDays_ = new XsdTypeNonNegativeInteger(businessDaysNode);
                }
            }
            
        
        }
        
    
        #region BusinessDays_
        private XsdTypeNonNegativeInteger businessDays_;
        public XsdTypeNonNegativeInteger BusinessDays_
        {
            get
            {
                if (this.businessDays_ != null)
                {
                    return this.businessDays_; 
                }
                else if (this.businessDaysIDRef_ != null)
                {
                    businessDays_ = IDManager.getID(businessDaysIDRef_) as XsdTypeNonNegativeInteger;
                    return this.businessDays_; 
                }
                else
                {
                      return this.businessDays_; 
                }
            }
            set
            {
                if (this.businessDays_ != value)
                {
                    this.businessDays_ = value;
                }
            }
        }
        #endregion
        
        public string businessDaysIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-cd-5-4/SingleValuationDate.cpp
// SingleValuationDate.cpp 
#include "SingleValuationDate.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

SingleValuationDate::SingleValuationDate(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //businessDaysNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* businessDaysNode = xmlNode->FirstChildElement("businessDays");

   if(businessDaysNode){businessDaysIsNull_ = false;}
   else{businessDaysIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- businessDaysNode , address : " << businessDaysNode << std::endl;
   #endif
   if(businessDaysNode)
   {
      if(businessDaysNode->Attribute("href") || businessDaysNode->Attribute("id"))
      {
          if(businessDaysNode->Attribute("id"))
          {
             businessDaysIDRef_ = businessDaysNode->Attribute("id");
             businessDays_ = boost::shared_ptr<XsdTypeNonNegativeInteger>(new XsdTypeNonNegativeInteger(businessDaysNode));
             businessDays_->setID(businessDaysIDRef_);
             IDManager::instance().setID(businessDaysIDRef_,businessDays_);
          }
          else if(businessDaysNode->Attribute("href")) { businessDaysIDRef_ = businessDaysNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { businessDays_ = boost::shared_ptr<XsdTypeNonNegativeInteger>(new XsdTypeNonNegativeInteger(businessDaysNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeNonNegativeInteger> SingleValuationDate::getBusinessDays()
{
   if(!this->businessDaysIsNull_){
        if(businessDaysIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeNonNegativeInteger>(IDManager::instance().getID(businessDaysIDRef_));
        }else{
             return this->businessDays_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeNonNegativeInteger>();
   }
}
}

